java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/routes/RouteRelationExtractor.java

```diff
@@ -0,0 +1,603 @@
+import net.osmand.osm.io.OsmStorageWriter;
+import net.osmand.render.FindByRenderingTypesRules;
+import net.osmand.shared.io.KFile;
+import net.osmand.util.Algorithms;
+import net.osmand.util.MapUtils;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorOutputStream;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.xmlpull.v1.XmlPullParserException;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.*;
+import java.nio.file.Files;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import static net.osmand.IndexConstants.GPX_GZ_FILE_EXT;
+import static net.osmand.gpx.GPXUtilities.OSMAND_EXTENSIONS_PREFIX;
+import static net.osmand.gpx.GPXUtilities.writeNotNullText;
+import static net.osmand.obf.OsmGpxWriteContext.OSM_TAG_PREFIX;
+import static net.osmand.router.RouteExporter.OSMAND_ROUTER_V2;
+
+public class RouteRelationExtractor {
+//	private boolean DEBUG = false;
+	private static final Log log = LogFactory.getLog(RouteRelationExtractor.class);
+	int countFiles;
+	int countWays;
+	int countNodes;
+	DBDialect osmDBdialect = DBDialect.SQLITE;
+	private final double precisionLatLonEquals = 1e-5;
+
+	private String[] customStyles = {
+			"default.render.xml",
+			"routes.addon.render.xml"
+			// "skimap.render.xml" // could work instead of default.render.xml but not together
+	};
+	private static final Map<String, String> customProperties = Map.of(
+			// routes.addon.render.xml
+			"showCycleRoutes", "true",
+			"showMtbRoutes", "true",
+			"hikingRoutesOSMC", "walkingRoutesOSMC",
+			"showDirtbikeTrails", "true",
+			"horseRoutes", "true",
+			"showFitnessTrails", "true",
+			"showRunningRoutes", "true",
+			"whiteWaterSports", "true" // default.render.xml
+			// "pisteRoutes", "true" // skimap.render.xml conflicts with default
+	);
+	private FindByRenderingTypesRules finder = new FindByRenderingTypesRules(customStyles, customProperties);
+	private final String[] filteredTags = {
+			"-bus",
+			"-road",
+			"hiking",
+			"bicycle",
+			"foot",
+			"-ferry",
+			"mtb",
+			"-power",
+			"-railway",
+			"piste",
+			"-train",
+			"ski",
+			"-tram",
+			"-detour",
+			"-tracks",
+			"-trolleybus",
+			"horse",
+			"-share_taxi",
+			"-subway",
+			"-emergency_access",
+			"snowmobile",
+			"historic",
+			"running",
+			"fitness_trail",
+			"-light_rail",
+			"canoe",
+			"-pipeline",
+			"canyoning",
+			"junction",
+			"motorboat",
+			"-yes",
+			"waterway",
+			"boat",
+			"worship",
+			"aerialway",
+			"inline_skates",
+			"transhumance",
+			"historic_railway",
+			"via_ferrata",
+			"-monorail",
+			"hiking;mtb",
+			"-funicular" // count 200
+			// "walking" count 164
+	};
+
+	public static void main(String[] args) {
```

Comment:
0. Init MapRenderingTypesEncoder (transform tags) and _Rendering_ class (separate) (for get icon, shields, etc.)
1. Glue way and node from relation to List<Way> and List<Node>. Will use in OBF generation in the future.
2. Get BinaryMapDataObject (or similar object) for these List<Way> and List<Node> (release in MapRenderingTypesEncoder)
3. From _Rendering_ get icon, get shield, get textColor etc. 
4. Write to GPX things

Reply:
0,3. done
1,2. todo (next task)
4. unknown